lib/api/apiUtils/bucket/bucketDeletion.js

```diff
@@ -151,12 +152,23 @@ export function deleteBucket(bucketMD, bucketName, canonicalID, log, cb) {
                     return _deleteMPUbucket(bucketName, log, next);
                 });
         },
+        function destroyBucketKey(next) {
```

Comment:
I think this step should only be done after the actual bucket is destroyed.  If we destroy the key at this step and the bucket is not actually destroyed, then if we "invisibly recreate" the flagged bucket on an object put for instance, we will be missing the key.